exonum/src/runtime/mod.rs

```diff
@@ -438,6 +438,30 @@ impl<T: Runtime> From<T> for Box<dyn Runtime> {
     }
 }
 
+/// Specifies system identifier for a [`Runtime`].
+pub trait WellKnownRuntime: Runtime {
+    /// Identifier of this runtime.
+    const ID: u32;
+}
+
+#[derive(Debug)]
+/// Instance of [`Runtime`] with corresponding ID.
```

Comment:
For the potential question of why I introduced this entity - implementing `WellKnownRuntime` for `(u32, Box<dyn Runtime>)` breaks the Rust's orphan rule.